spring-cloud-gcp-pubsub-stream-binder/src/main/java/com/google/cloud/spring/stream/binder/pubsub/provisioning/PubSubChannelProvisioner.java

```diff
@@ -132,6 +138,6 @@ private Topic makeSureTopicExists(String topicName, boolean autoCreate) {
 			}
 		}
 
-		return topic;
+		return Optional.ofNullable(topic);
```

Comment:
I don't think the topic could ever be null at this point. If it's null, and we don't auto-create, an exception will be thrown.

Reply:
I thought so too, but take a look at the comment above the `LOGGER` line - it can, in fact, be null, and one the unit tests  failed for me when I assumed it always to be non-null.

Reply:
That has something to do with concurrency. Which test?

Reply:
Right, but it could still return `null`, however unlikely. Hence, the `Optional`

Reply:
It doesn't look like we even use the returned object anywhere. We might as well return nothing, and make the method just that tiny bit more efficient.

Reply:
I ended up having to use it in the validation check that ensures the subscription & topic match.

Reply:
I think this whole situation with the topic potentially being created concurrently and being null is error-prone. Maybe the way to handle that `AlreadyExistsException` is to just try to fetch the topic instead of ignoring it. WDYT?

Reply:
It's a very small chance of it happening, and if it does happen, it just means that a validation check is skipped. Wrapping it in an optional is a clean way to ensure the code doesn't throw any unexpected NPEs.

Reply:
I think it's just more confusing to the caller of the method, but we can take care of it in a separate PR.